fwsrc/uart.c

```diff
@@ -1,28 +1,52 @@
-/******************************************************************************
- * Copyright 2013-2014 Espressif Systems (Wuxi)
- * Copyright 2015 <>< Charles Lohr
+/*
```

Comment:
I pulled this src in from SDK 3.0, then readded the charrx() function call, It happens from a task now instead of directly from the interrupt!